Assets/Scripts/Actors/ActorComponents/ActorPhysics.cs

```diff
@@ -319,54 +320,7 @@ public void ClimbSurface( Vector3 movement )
 		}
 	}
 
-	public void StopClimbing()
-	{
-		if ( _climbSurface )
-		{
-			_climbSurface = null;
-			_climbTag = null;
-			rigidbody.useGravity = true;
-			lifter.gameObject.SetActive( true );
-			bumper.gameObject.SetActive( true );
-			climbBumper.gameObject.SetActive( false );
-
-			ChangeState( ActorStates.Jumping );
-
-			if ( !actor.animator )
-			{
-				actor.animator.SetBool( "isClimbing", false );
-			}
-		}
-	}
-
-	public void RollCheck()
-	{
-		if ( IsInState( ActorStates.Rolling ) )
-		{
-			if ( _rollCooldownTimer >= _rollTime )
-			{
-				ChangeState( ActorStates.Jumping );
-				actor.animator.SetBool( "isRolling", false );
-			}
-		}
-		else if ( Input.GetButtonDown( "Roll" + PlatformUtils.platformName ) &&
-		          _rollCooldownTimer >= _rollCooldownTime && !_moveVec.IsZero() )
-		{
-			ChangeState( ActorStates.Rolling );
-			actor.animator.SetBool( "isRolling", true );
-
-			_rollCooldownTimer = 0f;
-		}
-
-		_rollCooldownTimer += Time.deltaTime;
-	}
-
-	public void RollMovement()
-	{
-		MoveAtSpeed( _moveVec.normalized, _rollMoveSpeed );
-	}
-
-	public void JumpCheck()
+	public bool JumpCheck()
```

Comment:
I see what you mean, but I think it's consistent within the `ActorPhysics` system. All of the `XCheck()` methods determine if action X can happen, perform the necessary actions as needed, and then return the result of the attempt. I think the bigger problem is that we're both checking if the action is possible and performing the action in the same method. With the latest round of changes the more appropriate way of doing this would be to have a separate `StartJumping()` method, something in the ilk of `startClimbing()` and `StartGliding()`.